### PR TITLE
[3.3.5] .learn adjustments

### PR DESCRIPTION
**Pull request by Treeston**
I've had this one in the back of my head for a while.

* [x] Add `.learn my quests` to learn quest-granted spells.
* [x] Adjust `.learn my spells` to use trainer system.
* [x] Adjust `.learn my talents` to use talents system properly.
* [x] Add `.learn debug` to learn a hand-picked list of spells for encounter testing.
* [x] Redesign the subcommand layout for `.learn`, because it is a bit of a mess right now.

### CODE REVIEW

File: src/server/game/Globals/ObjectMgr.h [hunk 1]
```diff
@@ -1490,6 +1490,7 @@ class TC_GAME_API ObjectMgr
         bool DeleteGameTele(std::string_view name);
 
         Trainer::Trainer const* GetTrainer(uint32 creatureId) const;
+        auto GetClassTrainers(uint8 classId) const { return Trinity::Containers::MapEqualRange(_classTrainers, classId); }
```
Comment (Shauren) [inline src/server/game/Globals/ObjectMgr.h:1493]: nope

Comment (Treeston) [reply src/server/game/Globals/ObjectMgr.h:1493]: Do you really want me to write out `Trinity::IteratorPair<WhateverTheFuckThatIteratorTypeIs>`?

Comment (Treeston) [reply src/server/game/Globals/ObjectMgr.h:1493]: `Trinity::IteratorPair<std::unordered_multimap<uint8, Trainer const*>::const_iterator>`

File: src/server/scripts/Commands/cs_learn.cpp [hunk 2]
```diff
@@ -139,58 +139,50 @@ class learn_commandscript : public CommandScript
...
+        auto trainers = sObjectMgr->GetClassTrainers(player->GetClass());
 
-            // skip spells with first rank learned as talent (and all talents then also)
-            if (GetTalentSpellCost(spellInfo->GetFirstRankSpell()->Id) > 0)
-                continue;
+        bool hadNew;
+        do
+        {
+            hadNew = false;
+            for (auto [_, trainer] : trainers)
```
Comment (Keader) [inline src/server/scripts/Commands/cs_learn.cpp:162]: let me use your knowledge a little bit?
what means this [_ stuff?

Comment (Treeston) [reply src/server/scripts/Commands/cs_learn.cpp:162]: `trainers` is `unordered_multimap<uint8, Trainer const*>`

The `uint8` (class ID) binds to unused local `_`, the `Trainer const*` binds to `trainer`.

Comment (Keader) [reply src/server/scripts/Commands/cs_learn.cpp]: oh... nice, ty <3

**Pull request merged**